modules/op-creating-pipeline-run-using-pipelines-as-code.adoc

```diff
@@ -0,0 +1,176 @@
+
+[discrete]
+.Matching a pull event to a pipeline run
+
+You can use the following example to match the `pipeline-pr-main` pipeline with a `pull_request` event that targets the `main` branch:
+
+[source,yaml]
+----
+...
+  metadata:
+    name: pipeline-pr-main
+  annotations:
+    pipelinesascode.tekton.dev/on-target-branch: "[main]" <1>
+    pipelinesascode.tekton.dev/on-event: "[pull_request]"
+...
+----
+<1> You can specifiy multiple branches by adding comma-separated entries. For example, `"[main, release-nightly]"`. In addition, you can specify the following:
+* Full references to branches such as `"refs/heads/main"`
+* Globs with pattern matching such as `"refs/heads/\*"`,
+* Tags such as `"refs/tags/1.\*"`.
+
+[discrete]
+.Matching a push event to a pipeline run
+
+You can use the following example to match the `pipeline-push-on-main` pipeline with a `push` event targeting the `refs/heads/main` branch:
+
+[source,yaml]
+----
+...
+  metadata:
+    name: pipeline-push-on-main
+  annotations:
+    pipelinesascode.tekton.dev/on-target-branch: "[refs/heads/main]" <1>
+    pipelinesascode.tekton.dev/on-event: "[push]"
+...
+----
+<1> You can specifiy multiple branches by adding comma-separated entries. For example, `"[main, release-nightly]"`. In addition, you can specify the following:
+* Full references to branches such as `"refs/heads/main"`
+* Globs with pattern matching such as `"refs/heads/\*"`,
+* Tags such as `"refs/tags/1.\*"`.
+
+[discrete]
+.Advanced event matching
+
+{pac} supports using Common Expression Language (CEL) based filtering for advanced event matching. If you have the `pipelinesascode.tekton.dev/on-cel-expression` annotation in your pipeline run, {pac} uses the CEL expression and skips the `on-target-branch` annotation. Compared to the simple `on-target-branch` annotation matching, the CEL expressions allows complex filtering and negation.
+
+To use CEL based filtering with {pac}, consider the following examples of annotations:
+
+* To match a `pull_request` event targeting the `main` branch and coming from the `wip` branch:
++
+[source,yaml]
+----
+...
+  pipelinesascode.tekton.dev/on-cel-expression: |
+    event == "pull_request" && target_branch == "main" && source_branch == "wip"
+...
+----
+
+* To run a pipeline only if a path has changed, you can use the `.pathChanged` suffix function with a glob pattern:
++
+[source,yaml]
+----
+...
+  pipelinesascode.tekton.dev/on-cel-expression: |
+    event == "pull_request" && "docs/\*.md".pathChanged() <1>
+...
+----
+<1> Matches all markdown files in the `docs` directory.
+
+* To match all pull requests starting with the title `[DOWNSTREAM]`:
++
+[source,yaml]
+----
+...
+  pipelinesascode.tekton.dev/on-cel-expression: |
+    event == "pull_request && event_title.startsWith("[DOWNSTREAM]")
+...
+----
+
+* To run a pipeline on a `pull_request` event, but skipping the `experimental` branch:
```

Comment:
```suggestion
* To run a pipeline on a `pull_request` event, but skip the `experimental` branch:
```
@sounix000 Check the technical accuracy.

Reply:
Comment addressed!

Reply:
I believe we can use - while

* To run a pipeline on a `pull_request` event, while skipping the `experimental` branch:

@Srivaralakshmi What do you think?

Reply:
Leaving it as is.

**Reason:** The first sentence suggests the action of skipping the experimental branch is ongoing, while the second sentence implies a one-time action.

@Srivaralakshmi Do you agree?